tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_reactive_ms.py

```diff
@@ -151,10 +160,19 @@ def check_automated_recovery_from_terminated_node(nodes):
 
     new_wnode = wait_for_new_worker_node_ipi(machineset, old_wnodes)
 
-    wait_for_osd_ids_come_up_on_node(new_wnode.name, old_osd_pod_ids, timeout=300)
-    log.info(
-        f"the osd ids {old_osd_pod_ids} Successfully come up on the node {new_wnode.name}"
-    )
+    # Wait for correct OSD distribution. This will also wait for the necessary number of OSD pods to be Running
+    retry((AssertionError), tries=10, delay=60)(verify_osd_distribution_on_provider)
+
+    # Verify that OSD pods are running on the new node
+    osd_ids_on_node = get_node_osd_ids(new_wnode.name)
+    assert osd_ids_on_node, f"No OSD pod is running on the node {new_wnode.name}."
+    log.info(f"OSD {osd_ids_on_node} are running on the node {new_wnode.name}.")
+
+    # The OSD IDs that were present on the terminated should be in running state now
+    osds_with_old_id = get_osd_pods_having_ids(old_osd_pod_ids)
+    assert len(set(osds_with_old_id)) == len(
+        set(old_osd_pod_ids)
+    ), f"One or more of the OSD IDs {old_osd_pod_ids} which were running on the terminated node are not present now"
```

Comment:
Same here.

Reply:
@yitzhak12 Wait is not required at this point because the verification to check the running status of OSDs and its distribution is already done. At this point the OSDs will be in state Running. So wait is not needed. Without wait we can catch an issue if OSDs is going back to non- running state.

Reply:
Okay, that makes sense.